### PR TITLE
feat: support go 1.16 and apple silicon

### PR DESCRIPTION
**Pull request by caarlos0**
closes https://github.com/goreleaser/goreleaser/issues/1952

will only be merged once go 1.16 is out.

### CODE REVIEW

**Review by radeksimko — APPROVED** [review]
LGTM, I tried using it to generate a compatible `darwin/arm64` build of `terraform-ls`:
https://github.com/hashicorp/terraform-ls/issues/349

Admittedly I don't have the new MacBook to test that it actually works, but compilation was successful.

The only question I would have is: What happens when `goreleaser release` runs under Go 1.15 or lower? I'm assuming this would gracefully fail, right?

**Comment by caarlos0** [comment]
> The only question I would have is: What happens when `goreleaser release` runs under Go 1.15 or lower? I'm assuming this would gracefully fail, right?

yes, go build will fail on the build step...

**Comment by bouk** [comment]
It would be good at add arm64 as a default here: https://github.com/goreleaser/goreleaser/blob/264dedc12a7ecab5a7ce5592e9e4aac1ca0494f6/internal/builders/golang/build.go#L52

**Comment by caarlos0** [comment]
> It would be good to add arm64 as a default here

done in #1999

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -34,7 +34,8 @@ jobs:
         name: Set up Go
         uses: actions/setup-go@v2
         with:
-          go-version: 1.15
+          stable: false
+          go-version: '1.16.0-beta1'
```
Comment (radeksimko) [inline .github/workflows/build.yml:38]:
```suggestion
          go-version: '1.16'
```

**Review by radeksimko — APPROVED** [review]
1.16 was just released:
https://golang.org/doc/go1.16

I think the only last missing piece of the puzzle is 1.16 CircleCI Docker image (which AFAIK is used for generating docs):
https://hub.docker.com/r/circleci/golang/tags?page=1&ordering=last_updated&name=1.16

**Review by crazy-max — APPROVED** [review]
LGTM minus upgrading GHA workflow to use Go 1.16.

**Pull request merged**